src/Content/Smilies.php

```diff
@@ -152,6 +152,34 @@ public static function getList(): array
 		return $params;
 	}
 
+	/**
+	 * Finds all used smilies (like :heart: or :p) in the provided text.
+	 *
+	 * @param string $text that might contain smilie usages (denoted by a starting colon)
+	 * @param bool   $extract_url whether to further extract image urls
+	 * @return array with smilie codes (colon included) as the keys, the smilie images as values
+	 */
+	public static function extractUsedSmilies(string $text, bool $extract_url = false): array
+	{
+		$emojis = [];
+
+		$smilies = self::getList();
+		$icons = $smilies['icons'];
+		foreach ($smilies['texts'] as $i => $name) {
+			if (strstr($text, $name)) {
```

Comment:
Due to the way some Friendica smilie code are defined with legacy text smileys (like `:)`, `;)`, `:p`), a simple search of the string in the body is likely to produce false positives. I'd prefer if you performed a regular expression search verifying that there aren't any word characters on either side of the smilie code.

Additionally, if there are specific BBCode tags in the body (`[nosmilies]` or the content of `[code]` tags), smilies shouldn't be searched. Please have a look at the existing uses of `Util\Strings::performWithEscapedTags()`.

Reply:
Thank you! However, currently `Smilies::replace` does not seem to handle word boundaries either:
![image](https://github.com/friendica/friendica/assets/14026120/9223c748-4696-4dd0-9bbc-8e6366aa524d)
Is it expected or maybe I should fix that as well?